### ISSUE TITLE
Kiali causing errors 503 and 504 at EKS apiserver.

### ISSUE THREAD

**Issue by sergiomacedo** [kiali/kiali#3933]
**Describe the bug**
After instaling kiali, multiple 504 errors are show, at seemingly random intervals, for LIST verbs refering to multiple istio objects like gateways,virtualservices,peerauthentication,destinationrules etc. 

This problem, besides causing alarms on our monitoring system (which can be adjusted), also seems to cause authentication problems with EKS. Perhaps overloading the apiserver?

I was able to reproduce the scenario on two different clusters, in different AWS accounts.

Additional information: our clusters have 70+ namespaces but, as of now, istio is enabled only in a handful (fewer than 10) of namespaces. However, kiali operator set the label "kiali.io/member-of" on all 70+ of them.

**Versions used**
Kiali: v1.33.1 (179cd6b016cd15deac16266520bb406185508b74)
Istio: 1.8.5
Kubernetes flavour and version: v1.16.15-eks-ad4801

**To Reproduce**
1. Install prometheus+grafana and enable cluster monitoring
2. Install istio
2. Install kiali
3. Go to prometheus or grafana endpoints and execute/add a panel with the following query:

`sum by(resource, subresource, verb, code) (rate(apiserver_request_total{code=~"5..",job="apiserver"}[5m])) / sum by(resource, subresource, verb, code) (rate(apiserver_request_total{job="apiserver"}[5m]))  > 0.05`

4. Let it running for a while and you'll see multiple errors for verb LIST

![Screenshot from 2021-04-22 18-37-07](https://user-images.githubusercontent.com/10211244/115788286-cf68ae80-a399-11eb-87db-957911016093.png)

**Expected behavior**

I kiali shouldn't generate errors when trying to list objects it needs to watch.

Extra information:
**Kiali CR**

```
apiVersion: kiali.io/v1alpha1
kind: Kiali
metadata:
  name: kiali
  namespace: istio-system
  annotations:
    ansible.sdk.operatorframework.io/verbosity: "1"
spec:
  auth:
    strategy: "anonymous"
  deployment:
    view_only_mode: true
    ingress_enabled: false
  external_services:
    tracing:
      in_cluster_url: "http://tracing.istio-system/"
      url: "http://tracing.homolog.my.domain/"
      use_grpc: false
```
Note: grpc is disabled because I was not able to find the correct endpoint (adding or removing /jaeger had no effect).

Please let me know if you need more information.

**UPDATE***

With these values below, only the 503 errors appear on our monitoring.

```
  deployment:
    # Limits to 22 namespaces
    accessible_namespaces:
    - ns-group1-.*
    - ns-group2-.*
    ingress_enabled: false
    logger:
      log_level: debug
    view_only_mode: true
  external_services:
    custom_dashboards:
      discovery_enabled: "false"
  kubernetes_config:
    burst: 50
    cache_duration: 600
    cache_token_namespace_duration: 60
    qps: 10
```

**Comment by jmazzitelli** [comment]
> However, kiali operator set the label "kiali.io/member-of" on all 70+ of them.

This part is not a bug. You told Kiali it has access to all 70+ namespaces and so labeled them all as such. If you don't want that, don't tell Kiali it has access to them. You do this via [the `spec.deployment.accessible_namespaces` setting](https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml#L170-L178) which you did not set, so Kiali by default is given access to all of them minus some like namespaces that match the regex `kube.*`.

Please read [this section of the documentation](https://kiali.io/documentation/latest/installation-guide/#_namespace_management), and then configure Kiali correctly (i.e. give it access to only those namespaces you want to give it access to).

**Comment by sergiomacedo** [comment]
I read this part of the documentation. However, the plan is to deploy kiali with access to all 70+ namespaces (each one a different microservice for our system). So, limit the number of observable namespaces now will only delay the problem as we move to production (this is happening on our development cluster).

My apologies for not letting it clear on the opening text.

**Comment by sergiomacedo** [comment]
I reduced the number of accessible_namespaces to 23 namespaces and the problem still occurs, as you can see on the image below.
![Screenshot from 2021-04-22 20-26-26](https://user-images.githubusercontent.com/10211244/115797087-7274f480-a3a9-11eb-8411-b7f36f56ea04.png)

**Comment by jmazzitelli** [comment]
A [504 error is a Gateway Timeout error](https://tools.ietf.org/html/rfc7231#section-6.6.5). It sounds like Kiali is attempting to simply query the k8s master API for details about Istio objects in those 70+ namespaces (this is the LIST verb requests that you see referring to multiple istio objects like gateways,virtualservices,peerauthentication,destinationrules etc). and the requests are failing because the k8s master API cannot successfully process the requests.

So Kiali is simply looking for Istio objects in the namespaces it has access to (spec.deployment.accessible_namespaces) but your cluster or network is not able to handle the requests - hence 504 errors. I don't see how this is a "bug" in Kiali. It's a cluster problem not able to handle requests. That you reduced the number of namespaces to 23 and still see the error tells me something is not right in that cluster. The small amount of traffic generated by Kiali for querying 23 namespaces should never cause the server to fail multiple 504 errors.

Take a look in the Kiali pod logs and see what kind of errors you see in there. Maybe that could provide more details; but I doubt it. A 504 indicates a problem on the remote side (not the client side).

**Comment by sergiomacedo** [comment]
I understand your point. I thought it might be a bug in kiali's way to build this queries.

 I'll take a look at the pod's logs and also open a ticket with AWS. 
 
 And I'd thank you very much if you can think of any other test I might do to pinpoint the exact issue, being it network, request size or whatever.

**Comment by sergiomacedo** [comment]
@jmazzitelli the pod's logs returned these messages.

```
E0423 15:35:49.017174       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: Timeout: Too large resource version: 107015886, current: 107003075
I0423 15:36:41.671505       1 trace.go:205] Trace[654365506]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167 (23-Apr-2021 15:35:58.635) (total time: 43035ms):
Trace[654365506]: [43.035506504s] [43.035506504s] END
E0423 15:36:41.671527       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: Timeout: Too large resource version: 107016490, current: 107005130
I0423 15:36:48.625273       1 trace.go:205] Trace[272059564]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167 (23-Apr-2021 15:36:05.582) (total time: 43042ms):
Trace[272059564]: [43.042697001s] [43.042697001s] END
E0423 15:36:48.625297       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: Timeout: Too large resource version: 107015886, current: 107003075
I0423 15:38:10.132818       1 trace.go:205] Trace[939242510]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167 (23-Apr-2021 15:37:27.062) (total time: 43070ms):
Trace[939242510]: [43.070415205s] [43.070415205s] END
E0423 15:38:10.132840       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: Timeout: Too large resource version: 107015886, current: 107003075
I0423 15:38:13.298235       1 trace.go:205] Trace[1320386506]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167 (23-Apr-2021 15:37:37.268) (total time: 36030ms):

```
 
 The excessive time to complete the request is obvious and  the "Timeout" message may be related to this: [https://github.com/kubernetes/kubernetes/issues/82428](https://github.com/kubernetes/kubernetes/issues/82428)
 
Have you ever met this issue before?

**Comment by jmazzitelli** [comment]
> Have you ever met this issue before?

I have never seen this before.

**Comment by sergiomacedo** [comment]
@jmazzitelli I opened a case with AWS and at the same time I changed my kiali CR to these settings, but I'm worried about any possible side effects. Would you please share your thoughts on this?

```
  deployment:
    accessible_namespaces:
    - ns-group1-.*
    - ns-group2-.*
    ingress_enabled: false
    logger:
      log_level: debug
    view_only_mode: true
  external_services:
    custom_dashboards:
      discovery_enabled: "false"
    tracing:
      in_cluster_url: http://tracing.istio-system/
      url: http://tracing.homolog.internal.domain/
      use_grpc: false
  kubernetes_config:
    burst: 50
    cache_duration: 600
    cache_token_namespace_duration: 60
    qps: 10

```

ns-group1 and ns-group2 reduces the number of namespaces to 22, but the final configuration surely will be more than that.

**Comment by jmazzitelli** [comment]
That `spec.deployment.accessible_namespaces` setting looks fine to me. 

Just as a side note, [new performance enhancements](https://github.com/kiali/kiali-operator/pull/293) will be going into Kiali soon - these enhancements will be in the next release of Kiali (v1.34 coming in two weeks). I don't think it will do anything with respect to your 504 error, but it will help speed up the operator processing time when you have a large number of namespaces in accessible_namespaces (hundreds or more).

**Comment by sergiomacedo** [comment]
This is great news, @jmazzitelli. Do you have any suggestion of which other settings I should tune to reduce the load on the eks endpoint? 

Right now, with the values I posted above, the only errors are 503 CONNECT to the eks endpoint. AWS support is being extremely helpful... telling me to contact "kiali or istio people"... :(

**Comment by jmazzitelli** [comment]
> Right now, with the values I posted above, the only errors are 503 CONNECT to the eks endpoint.

I don't know the reason for the 503 - look at the kiali pod logs to see if you can get insight into where the error is occurring.

BTW: `503 Connect` is different than `504 Timeout` - so that would mean you have to look at something else. The title of this bug report is `Kiali causing error 504 at EKS apiserver` - did that go away? If so, please close this issue and remove the "bug" label, and [start a new Q&A discussion](https://github.com/kiali/kiali/discussions/categories/q-a) about this new problem which also sounds like a misconfiguration and not a bug.

**Comment by sergiomacedo** [comment]
@jmazzitelli this issue seems similar. The message @FilipB sees is the same I do with the default QPS/Burst parameters.

#3957 https://github.com/kiali/kiali/issues/3957

**Comment by sergiomacedo** [comment]
Ok... This is new... 

```

2021-04-28T19:01:09Z WRN GetPodProxyStatus is failing for [namespace: ns-group1-app1] [pod: ns-group1-app1-workers-547767577b-c2vd2]: unable to proxy Istiod pods. Make sure your Kubernetes API server has access to the Istio control plane through 8080 port 
2021-04-28T19:01:09Z WRN GetPodProxyStatus is failing for [namespace: ns-group1-app2] [pod: ns-group1-app2-89d68b8d6-rqczg]: unable to proxy Istiod pods. Make sure your Kubernetes API server has access to the Istio control plane through 8080 port  
2021-04-28T19:01:09Z WRN GetPodProxyStatus is failing for [namespace: ns-group1-app2] [pod: ns-group1-app2-monitors-7b9fbc6c6f-x22qd]: unable to proxy Istiod pods. Make sure your Kubernetes API server has access to the Istio control plane through 8080 port 
2021-04-28T19:01:09Z WRN GetPodProxyStatus is failing for [namespace: ns-group1-app2] [pod: ns-group1-app2-monitors-7b9fbc6c6f-x22qd]: unable to proxy Istiod pods. Make sure your Kubernetes API server has access to the Istio control plane through 8080 port 
2021-04-28T19:01:10Z WRN GetPodProxyStatus is failing for [namespace: ns-group1-app2] [pod: ns-group1-app2-monitors-7b9fbc6c6f-x22qd]: unable to proxy Istiod pods. Make sure your Kubernetes API server has access to the Istio control plane through 8080 port 
2021-04-28T19:01:10Z WRN GetPodProxyStatus is failing for [namespace: ns-group1-app2] [pod: ns-group1-app2-monitors-7b9fbc6c6f-x22qd]: unable to proxy Istiod pods. Make sure your Kubernetes API server has access to the Istio control plane through 8080 port 
2021-04-28T19:01:10Z WRN GetPodProxyStatus is failing for [namespace: ns-group1-app3] [pod: ns-group1-app3-api-7496985749-nfldn]: unable to proxy Istiod pods. Make sure your Kubernetes API server has access to the Istio control plane through 8080 port 
2021-04-28T19:01:10Z WRN GetPodProxyStatus is failing for [namespace: ns-group1-app2] [pod: ns-group1-app2-cc68b7757-nvr8m]: unable to proxy Istiod pods. Make sure your Kubernetes API server has access to the Istio control plane through 8080 port  

```

And it happened together with the 503 error... It surely makes sense....

**Comment by sergiomacedo** [comment]
If it helps, here are the messages that happen together with the 504 errors:

```
E0429 21:58:56.016672       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: Timeout: Too large resource version: 109397965, current: 109357860
I0429 22:00:34.722741       1 trace.go:205] Trace[1195424759]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167 (29-Apr-2021 21:59:51.684) (total time: 43038ms):
Trace[1195424759]: [43.038408547s] [43.038408547s] END
E0429 22:00:34.722770       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: Timeout: Too large resource version: 109397965, current: 109357860
I0429 22:01:52.207523       1 trace.go:205] Trace[149387991]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167 (29-Apr-2021 22:01:09.173) (total time: 43033ms):
Trace[149387991]: [43.03380152s] [43.03380152s] END
E0429 22:01:52.207557       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: Timeout: Too large resource version: 109397965, current: 109357860
I0429 22:03:35.237925       1 trace.go:205] Trace[1390882304]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167 (29-Apr-2021 22:02:52.198) (total time: 43039ms):
Trace[1390882304]: [43.03975923s] [43.03975923s] END
E0429 22:03:35.237954       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: Timeout: Too large resource version: 109397965, current: 109357860
I0429 22:04:56.483191       1 trace.go:205] Trace[534941433]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167 (29-Apr-2021 22:04:13.448) (total time: 43034ms):
Trace[534941433]: [43.034672894s] [43.034672894s] END
E0429 22:04:56.483221       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: Timeout: Too large resource version: 109397965, current: 109357860
```

**Comment by lucasponce** [comment]
> Right now, with the values I posted above, the only errors are 503 CONNECT to the eks endpoint. AWS support is being extremely helpful... telling me to contact "kiali or istio people"... :(

Ouch. Sorry, we'll try to help but our testing capabilities can be limited and we may need to use other platforms.

I guess you are using Istio 1.8.x in a Kubernetes 1.16.x version provided by EKS, some of the messages in the log may be due those differences.

Kiali 1.33.x are tested on Istio 1.9 and focused on higher Kubernetes 1.17 - 1.20.

I wonder if you could downgrade Kiali to the version which is shipped in Istio 1.8.x -> Kiali 1.26.x to see if at least you have an older Kiali version but stable.

If you can test that and tell us the result we can collect more info to debug and try to help.

Also, any comment is most that welcome, we'll try to reproduce but for large clusters scenarios we need to simulate in other platforms (not always testing the same combination).

**Comment by sergiomacedo** [comment]
You're right about eks and istio versions, @lucasponce. I'll test with an older kiali version to see if something changes.

**Comment by lucasponce** [comment]
@jmazzitelli what would be the best way for @sergiomacedo to install the Kiali 1.26.x ?

**Comment by jmazzitelli** [comment]
> I'm using helm to install kiali-operator on my cluster. However, it seems kiali-operator always tries to download the latest image.

When falling back to a previous version of the operator, you can explicitly tell the operator to use a specific image via: https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml#L250-L264

Not specifying this causes the operator to install the "lastrelease" as defined here: https://github.com/kiali/kiali-operator/blob/master/playbooks/default-supported-images.yml#L1

This brings up an interesting point that is completely unrelated to this issue, but one I need to start thinking about. We recently introduced a feature in which the operator will not allow you to set this image_version field (you will be required to install the version the operator has set by default). This clearly isn't going to work when you have an older operator but then Kiali releases a new version of the server which may require new/different permissions or CR settings. We may have to change that feature's behavior. I'll write a separate github issue for this... this is going to be a problem I think.

**Comment by jmazzitelli** [comment]
> I just thought it a bit confusing that a specific operator version would resort to the latest application by default. I would expect it to deploy the application with the same operator version.

@sergiomacedo there is (was) a reason for that - but I'm going to switch it back. Read [this issue](https://github.com/kiali/kiali/issues/3968) I just created if you care about the gory details :)

**Comment by sergiomacedo** [comment]
The HTTP 504 errors happen with every version above 1.28.

Here's the output of one, taken from cloudwatch:

```
Field | Value
-- | --
@ingestionTime | 1620534966163
@log | 129817111992:/aws/eks/k8s-homolog/cluster
@logStream | kube-apiserver-audit-77883bd60f5310e74ad3561d861b116b
@message | {"kind":"Event","apiVersion":"audit.k8s.io/v1","level":"Metadata","auditID":"xxxxx","stage":"ResponseComplete","requestURI":"/apis/networking.istio.io/v1alpha3/namespaces/namespace1/destinationrules?resourceVersion=112861907","verb":"list","user":{"username":"system:serviceaccount:istio-system:kiali-service-account","uid":"xxxxx","groups":["system:serviceaccounts","system:serviceaccounts:istio-system","system:authenticated"]},"sourceIPs":["x.y.z.203"],"userAgent":"Go-http-client/2.0","objectRef":{"resource":"destinationrules","namespace":"namespace1","apiGroup":"networking.istio.io","apiVersion":"v1alpha3"},"responseStatus":{"metadata":{},"status":"Failure","reason":"Timeout","code":504},"requestReceivedTimestamp":"2021-05-09T04:35:57.888129Z","stageTimestamp":"2021-05-09T04:36:00.888793Z","annotations":{"authorization.k8s.io/decision":"allow","authorization.k8s.io/reason":"RBAC: allowed by RoleBinding \"kiali/namespace1\" of Role \"kiali-viewer\" to ServiceAccount \"kiali-service-account/istio-system\""}}
@timestamp | 1620534961110
annotations.authorization.k8s.io/decision | allow
annotations.authorization.k8s.io/reason | RBAC: allowed by RoleBinding "kiali/namespace1" of Role "kiali-viewer" to ServiceAccount "kiali-service-account/istio-system"
apiVersion | audit.k8s.io/v1
kind | Event
level | Metadata
objectRef.apiGroup | networking.istio.io
objectRef.apiVersion | v1alpha3
objectRef.namespace | namespace1
objectRef.resource | destinationrules
requestReceivedTimestamp | 2021-05-09T04:35:57.888129Z
requestURI | /apis/networking.istio.io/v1alpha3/namespaces/namespace1/destinationrules?resourceVersion=112861907
responseStatus.code | 504
responseStatus.reason | Timeout
responseStatus.status | Failure
sourceIPs.0 | x.y.z.203
stage | ResponseComplete
stageTimestamp | 2021-05-09T04:36:00.888793Z
user.groups.0 | system:serviceaccounts
user.groups.1 | system:serviceaccounts:istio-system
user.groups.2 | system:authenticated
user.username | system:serviceaccount:istio-system:kiali-service-account
userAgent | Go-http-client/2.0
verb | list

```

This error occurred at night, on the development environment.

**Comment by lucasponce** [comment]
> requestURI | /apis/networking.istio.io/v1alpha3/namespaces/namespace1/destinationrules?resourceVersion=112861907
> responseStatus.code | 504
> responseStatus.reason | Timeout
> responseStatus.status | Failure

It's a little bit strange that error, perhaps you have reviewed, is it possible that in your EKS server there is some extra security service working like a firewall ?

**Comment by sergiomacedo** [comment]
@lucasponce  does these 504 errors may be related to how kiali "talks" to the kubernetes clusters? As I said, they start to happen if I try kiali 1.29 or newer... I'll not ask you to change anything on the newer versions, of course, but I'd like to be sure the problem is this so I can put it as another item being affected by our 1.16 eks clusters (istio itself is another one. I'm stuck with 1.8.x due the same reason).

**Comment by lucasponce** [comment]
@sergiomacedo I don't think it's related how Kiali "talks" with the cluster, then it should be easy seen in more cluster.

Kiali uses the deps
```
	core_v1 "k8s.io/api/core/v1"
	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/apimachinery/pkg/labels"
	"k8s.io/apimachinery/pkg/runtime"
	"k8s.io/apimachinery/pkg/types"
	"k8s.io/client-go/rest"
```

How could I reproduce what you are seeing ? I'd like to help, if you can also provide me some access to a demo cluster where the issue is present I can try to take a look and see if there is something missing.

But it looks there is something in the EKS side that is provoking this problem.

**Comment by sergiomacedo** [comment]
I was thinking about libary versions... Perhaps the change happened on these packages not on kiali itself. 

Do you have an AWS account available, to create an EKS 1.16 cluster, @lucasponce ?

Kiali showed the same behaviour on two of our clusters, in different accounts, networks, etc... The only things in common were EKS version (1.16) and AWS region (us-east-1), using . 

If you could be so kind and try to recreate this scenario on AWS... 

I know that this kubernetes version is near EOL but I believe this kind of research is important to avoid similar (or worst) things in the future...

**Comment by lucasponce** [comment]
@sergiomacedo I'll try, which installation procedure did you follow to get the same error ?
The issues is seen with an empty cluster and one of the demo apps on the demo profile ?

**Comment by sergiomacedo** [comment]
Terraform using helm_release as provider. I still haven't tried with an empty cluster. I'll do that next week.
The configuration below returns 100+ accessible namespaces to kiali.

```
resource "helm_release" "kiali" {
  name             = "kiali"
  chart            = "kiali-operator"
  repository       = "https://kiali.org/helm-charts"
  version          = var.kiali_version
  create_namespace = true
  namespace        = "kiali-operator"
  set {
    name  = "cr.create"
    value = "false"
  }
  set {
    name  = "cr.namespace"
    value = "istio-system"
  }
  depends_on = [
    helm_release.istiod,
  ]
}
```
```
resource "kubectl_manifest" "kiali-cr" {
  yaml_body  = <<YAML
apiVersion: kiali.io/v1alpha1
kind: Kiali
metadata:
  name: kiali
  namespace: istio-system
  annotations:
    ansible.sdk.operatorframework.io/verbosity: "1"
spec:
  auth:
    strategy: "anonymous"
  server:
    metrics_enabled: false
  kiali_feature_flags:
    istio_injection_action: false
    ui_defaults:
      refresh_interval: "1m"
  deployment:
    replicas: 1
    view_only_mode: true
    ingress_enabled: false
    image_version: "v${var.kiali_version}"
    logger:
      log_level: debug
  external_services:
    istio:
      component_status:
        enabled: true
        components:
        - app_label: istiod
          is_core: true
        - app_label: istio-ingressgateway
          is_core: true
        - app_label: istio-egressgateway
          is_core: false
    custom_dashboards:
      discovery_enabled: false
    tracing:
      use_grpc: false
      in_cluster_url: "http://tracing.istio-system/"
      url: "http://${var.jaeger-hostname}/"
    grafana:
      enabled: false
    prometheus:
      url: "http://prometheus-istio-server.istio-system:9090"
    istio:
      config_map_name: "istio-${var.revision}"
  istio_labels:
    injection_label_name: "istio.io/rev"
YAML
  depends_on = [helm_release.kiali]
}
```

**Comment by lucasponce** [comment]
Hi @sergiomacedo, 
I could get an AWS account; I need to check, but I'll be able to help more.
I suggest if we can test this in an incremental way; I have suspects that is a configuration/environmental variable, so I'd start with testing a basic cluster + demo profile first, and from that we can check other configs.

**Comment by sergiomacedo** [comment]
Hello @lucasponce. I believe we can put this issue on hold. I was just informed that we WILL have to upgrade our clusters to 1.17 in the next weeks. So, you can let it go for now. I'll revive this thread with new information as soon as at least our development cluster is updated.

**Comment by lucasponce** [comment]
Thanks @sergiomacedo, I've discussed with my colleagues that perhaps there is some step in the flow terraform -> helm-charts -> operators that is out of sync and it may be related.

In any case, when you can, ping me back, now that we have some AWS accounts we'll try to reproduce the issue.

**Comment by sergiomacedo** [comment]
It's highly likely we'll take our clusters at least to 1.18 in a short time frame, so any problems related to kubernetes version will stop being an issue... until k8s 1.21, of course... 😂😂😂😂😂😂

**Comment by lucasponce** [comment]
Thanks for the update @sergiomacedo; hope the upgrade goes well in your clusters, just let us know if the problems are gone or there is new data.

**Comment by sergiomacedo** [comment]
FYI @lucasponce, same behaviour with kiali 1.35.0 and EKS 1.17. I'll let you know as soon as we get to 1.18.

**Comment by xeviknal** [comment]
> 2021-04-28T19:01:09Z WRN GetPodProxyStatus is failing for [namespace: ns-group1-app1] [pod: ns-group1-app1-workers-547767577b-c2vd2]: unable to proxy Istiod pods. Make sure your Kubernetes API server has access to the Istio control plane through 8080 port 

This happens when your Kubernetes API doesn't have access to the control plane namespace. Therefore, opening the port 8080 from the KubeAPI namespace to the `istio-system` should resolve this problem.
There is another feature that needs the port 15000 opened too. So you might want to open that port by the way.

Check the FAQ entry we have with regards to this specific situation: https://kiali.io/documentation/latest/faq/#google-cloud-private-cluster

**Comment by sergiomacedo** [comment]
@lucasponce I can now confirm that the issue was related to the eks cluster version. It's been eight days since I upgraded our development cluster to 1.18 and five days since I upgraded kiali to 1.37.0 and no errors appeared on cloudwatch/grafana related to istio/kiali objects. 

I'm truly sorry this took so long and for opening it as a potential bug. Anyways, it's a very good warning regarding the matching of client libraries and kubernetes versions. 

Thanks for your attention, guys. This issue can now be closed.
